keps/sig-auth/3926-handling-undecryptable-resources/README.md

```diff
@@ -0,0 +1,952 @@
+  //          doing.
+  // WARNING: Vendors will most likely consider using this option to be breaking the
+  //          support of their product.
+  IgnoreStoreReadErrorWithClusterBreakingPotential bool
+}
+```
+
+#### Protecting unconditional deletion
+
+A "delete" verb on a resource is not usually considered a privileged action. As the previous
+section explains, deletion of a resource might carry unexpected consequences. Unconditional
+deletions should therefore have their own extra admission.
+
+The unconditional deletion admission:
+1. checks if a "delete" request contains the `IgnoreStoreReadErrorWithClusterBreakingPotential` option
+2. if it does, it checks the RBAC of the request's user for the `delete-ignore-read-errors` verb of the given resource
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+- [x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `k8s.io/apiserver/pkg/storage/etcd3`: `28.9.2023` - `77%`
+- `k8s.io/apimachinery/pkg/apis/meta/v1`: `28.9.2023` - `48.1%`
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
```

Comment:
Are you planning any integration and e2e tests? Those should be described here and in the e2e section below.

Reply:
I am planning integration tests at least but I assume e2e test might be helpful, too. I am not sure which package to add those to just yet though.

Reply:
No need for exact package, for alpha it's totally fine just to describe the intended tests.